src/univariate/continuous/generalizedpareto.jl

```diff
@@ -169,6 +169,189 @@ function quantile(d::GeneralizedPareto{T}, p::Real) where T<:Real
+    ξ = ss.ξ
+    return GeneralizedPareto(d.μ, ξ / d.θ, ξ)
+end
+
+#
+# empirical bayes
+#
+
+"""
+    GeneralizedParetoKnownMu(μ)
+
+Represents a [`GeneralizedPareto`](@ref) where ``\\mu`` is known.
+"""
+struct GeneralizedParetoKnownMu{T} <: IncompleteDistribution
+    μ::T
+end
+
+"""
+    fit(GeneralizedPareto, x; μ, kwargs...)
+
+Fit a [`GeneralizedPareto`](@ref) with known location `μ` to the data `x`.
+
+The fit is performed using the Empirical Bayes method of [^ZhangStephens2009][^Zhang2010].
+
+# Keywords
+- `sorted::Bool=issorted(x)`: If `true`, `x` is assumed to be sorted. If `false`, a sorted
+    copy of `x` is made.
+- `improved::Bool=true`: If `true`, use the adaptive empirical prior of [^Zhang2010].
+    If `false`, use the simpler prior of [^ZhangStephens2009].
+- `min_points::Int=30`: The minimum number of quadrature points to use when estimating the
+    posterior mean of ``\\theta = \\frac{\\xi}{\\sigma}``.
+
+[^ZhangStephens2009]: Jin Zhang & Michael A. Stephens (2009)
+                      A New and Efficient Estimation Method for the Generalized Pareto Distribution,
+                      Technometrics, 51:3, 316-325,
+                      DOI: [10.1198/tech.2009.08017](https://doi.org/10.1198/tech.2009.08017)
+[^Zhang2010]: Jin Zhang (2010) Improving on Estimation for the Generalized Pareto Distribution,
+              Technometrics, 52:3, 335-339,
+              DOI: [10.1198/TECH.2010.09206](https://doi.org/10.1198/TECH.2010.09206)
+"""
+function StatsBase.fit(::Type{<:GeneralizedPareto}, x::AbstractArray; μ::Real, kwargs...)
+    return fit(GeneralizedParetoKnownMu(μ), x; kwargs...)
+end
+function StatsBase.fit(g::GeneralizedParetoKnownMu, x::AbstractArray; kwargs...)
+    return fit_empiricalbayes(g, x; kwargs...)
+end
+
+# Note: our ξ is ZhangStephens2009's -k, and our θ is ZhangStephens2009's -θ
+
+function fit_empiricalbayes(
+    g::GeneralizedParetoKnownMu,
+    x::AbstractArray;
+    sorted::Bool=issorted(vec(x)),
+    improved::Bool=true,
+    min_points::Int=30,
+)
+    μ = g.μ
+    T = Base.promote_eltype(x, μ)
+    # fitting is faster when the data are sorted
+    xsorted = sorted ? vec(x) : sort(vec(x))
+    xmin, xmax = @inbounds xsorted[1], xsorted[end]
+    if xmin ≈ xmax
+        # support is nearly a point. solution is not unique; any solution satisfying the
+        # constraints σ/ξ ≈ 0 and ξ < 0 is acceptable. we choose the ξ = -1 solution, i.e.
+        # the uniform distribution
+        return GeneralizedPareto(μ, max(eps(zero(T)), xmax - μ), -one(μ))
+    end
+    # estimate θ using empirical bayes
+    θ_hat = _fit_gpd_θ_empirical_bayes(μ, xsorted, min_points, improved)
+    # estimate remaining parameters using MLE
+    return fit_mle(GeneralizedPareto, xsorted; μ=μ, θ=θ_hat)
+end
+
+# estimate θ̂ = ∫θp(θ|x,μ)dθ, i.e. the posterior mean using quadrature over grid
+# of minimum length `min_points + floor(sqrt(length(x)))` uniformly sampled over an
```

Comment:
Here it's because it's the method that paper used, and they empirically found that it worked well. I'd have to look at QuadGK more closely to get a sense for whether it would be more useful here.

Reply:
I'll add a flag for integrating with QuadGK instead of at the chosen quadrature points to evaluate the relative accuracy/speed for reasonable target distributions. If the QuadGK approach is universally better, then we can remove the explicit approach.

Reply:
So it turns out the QuadGK approach doesn't work here. Because this is a posterior mean, we need to compute the normalization constant of the posterior, which unfortunately QuadGK cannot do on log-scale, so it ends up evaluating to 0.